plugins/example-output/dummy.c

```diff
@@ -0,0 +1,56 @@
+#include <stdio.h>
+#include "suricata-plugin.h"
+#include "dummy.h"
+
+#define OUTPUT_NAME "plugin:dummy"
+
+typedef struct Context_ {
+} Context;
+
+static int dummy_write(const char *buffer, int buffer_len, LogFileCtx *ctx) {
+    printf("dummy_write\n");
+    printf("dummy_write: rotate=%d\n", ctx->rotation_flag);
+    return 0;
+}
+
+static void dummy_close(LogFileCtx *ctx) {
+    printf("dummy_close\n");
+    SCFree(ctx->plugin);
+}
+
+static int dummy_open(LogFileCtx *ctx, ConfNode *conf) {
+    printf("conf: %p\n", conf);
+    Context *context = SCCalloc(1, sizeof(Context));
+    ctx->Write = dummy_write;
+    ctx->Close = dummy_close;
+    ctx->plugin = context;
+    return 0;
+}
+
+/**
+ * Called by Suricata to initialize the module. This module registers
+ * new file type to the JSON logger.
+ */
+void sc_plugin_init(void)
```

Comment:
so the idea is that each plugin type has this common init function type and within this the plugin type specific setup will be done? (e.g. output, app-layer, etc)

Reply:
Correct. It would be a generic interface for any plugin. The plugin doesn't need to declare itself to be app-layer, output, or whatever. My feeling is that might create too much overhead, at least until we see patterns emerge.